src/Xamarin.Android.Build.Tasks/Utilities/AssemblyCompression.cs

```diff
@@ -67,18 +67,20 @@ public static CompressionResult Compress (AssemblyData data, string outputDirect
 			// }
 
 			data.DestinationPath = Path.Combine (outputDirectory, $"{Path.GetFileName (data.SourcePath)}.lz4");
-			data.SourceSize = (uint)fi.Length;
+			data.SourceSize = checked((uint)fi.Length);
 
+			int bytesRead;
 			byte[] sourceBytes = null;
 			byte[] destBytes = null;
 			try {
-				sourceBytes = bytePool.Rent (checked((int)fi.Length));
+				int fileSize = checked((int)fi.Length);
+				sourceBytes = bytePool.Rent (fileSize);
 				using (var fs = File.Open (data.SourcePath, FileMode.Open, FileAccess.Read, FileShare.Read)) {
-					fs.Read (sourceBytes, 0, (int)fi.Length);
+					bytesRead = fs.Read (sourceBytes, 0, fileSize);
 				}
 
-				destBytes = bytePool.Rent (LZ4Codec.MaximumOutputSize (sourceBytes.Length));
-				int encodedLength = LZ4Codec.Encode (sourceBytes, 0, checked((int)fi.Length), destBytes, 0, destBytes.Length, LZ4Level.L12_MAX);
+				destBytes = bytePool.Rent (LZ4Codec.MaximumOutputSize (bytesRead));
+				int encodedLength = LZ4Codec.Encode (sourceBytes, 0, bytesRead, destBytes, 0, destBytes.Length, LZ4Level.L12_MAX);
```

Comment:
So, I think this code previously was renting a larger `destBytes` byte array than needed. I think we probably had extra empty bytes at the end? Potentially garbage bytes, too?